agasc/scripts/update_obs_status.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env python
+"""
+Update the 'bad' and 'obs' tables in the agasc supplement (agasc_supplement.h5).
+
+This file is a supplement to the stable AGASC to inform star selection
+and star catalog checking.
+
+This script can add one or many bad stars to the bad star table, as well as observation status
+information relevant for the magnitude estimation from acq data.
+"""
+
+from astropy import table
+import yaml
+import pathlib
+import argparse
+from agasc.scripts.add_bad_star import add_bad_star
+import numpy as np
+import logging
+
+from agasc.supplement.magnitudes import star_obs_catalogs
+
+
+def parse_obs_status_file(filename):
+    """
+    Parse a yaml file and return a dictionary.
+
+    The dictionary will be of the form: {'obs': {}, 'bad': {}}
+
+    :param filename:
+    :return:
+    """
+    with open(filename) as fh:
+        status = yaml.load(fh, Loader=yaml.SafeLoader)
+    if 'obs' not in status:
+        status['obs'] = []
+    if 'bad' not in status:
+        status['bad'] = {}
+    for value in status['obs']:
+        obs = value['obsid']
+        from agasc.supplement.magnitudes import star_obs_catalogs
+        if star_obs_catalogs.STARS_OBS is None:
+            raise RuntimeError('Observation catalog is not initialized')
+
+        if 'agasc_id' not in value:
+            value['agasc_id'] = list(
+                sorted(star_obs_catalogs.STARS_OBS[star_obs_catalogs.STARS_OBS['obsid'] == obs]['agasc_id'])
+            )
+        else:
+            value['agasc_id'] = (list(np.atleast_1d(value['agasc_id'])))
+        if 'comments' not in value:
+            value['comments'] = ''
+
+    status['obs'] = {
+        (obs['obsid'], agasc_id): {
+            'ok': obs['ok'],
+            'comments': obs['comments']
+        }
+        for obs in status['obs'] for agasc_id in obs['agasc_id']
+    }
+
+    return status
+
+
+def parse_obs_status_args(filename=None, bad_star=[], bad_star_source=None,
+                          obs=None, status=None, comments='', agasc_id=None,
+                          **_
+                          ):
+    """
+    Combine obs/bad-star status from file and from arguments.
+
+    The arguments could be passed from and ArgumentParser doing::
+
+        parse_obs_status_args(vars(args))
+
+    :param filename: str
+    :param bad_star: int
+    :param bad_star_source: int
+    :param obs: int
+    :param status: int
+    :param comments: str
+    :param agasc_id: int
+    :return:
+    """
+    obs_status_override = {}
+    bad_star = (list(np.atleast_1d(bad_star)))
```

Comment:
I agree, but this convention came from the function for bad stars, where bad_star was an int, and I modified it so it can also be a list. I can change it though.